### ISSUE TITLE
Optional ultrafast COCO backend for the object-detection examples (after TorchMetrics support)

### ISSUE THREAD

**Issue by developer0hye** [huggingface/transformers#48698]
### Feature request

Would you welcome an optional `--coco_eval_backend ultrafast` in the two PyTorch object-detection examples, once the corresponding TorchMetrics backend is released? I maintain [ultrafast-pycocotools](https://github.com/developer0hye/ultrafast-pycocotools) and am requesting scope approval before opening a Transformers PR.

The change would be limited to `run_object_detection.py` and `run_object_detection_no_trainer.py`, their optional dependency instructions, and focused tests. The default remains pycocotools; preprocessing and model behavior are unchanged. The option routes through TorchMetrics' public `MeanAveragePrecision` backend API and is checked before dataset loading/training. Unsupported installations raise an actionable error, without silently falling back.

### Dependency and reviewable implementation

This depends on [Lightning-AI/torchmetrics#3500](https://github.com/Lightning-AI/torchmetrics/pull/3500), which is still a draft awaiting scope discussion. Released TorchMetrics **1.9.0 does not support this backend**; I verified the rejection in both examples. This is therefore a proposal for a follow-up integration, not a claim that installing the current releases is sufficient.

A reviewable implementation is available at [commit 5258ccf](https://github.com/developer0hye/transformers/commit/5258ccfc4d04031b831ecbf9295be0e5cd1b7469), based on upstream `ff4bb5e`. Its review-only `requirements-ultrafast.txt` pins the exact TorchMetrics PR commit and ultrafast-pycocotools >=0.1.11,<0.2. That Git dependency must be replaced with a released minimum before merging.

### Verification

- `HF_HUB_OFFLINE=1 OMP_NUM_THREADS=2 RAYON_NUM_THREADS=2 python -m pytest -q examples/pytorch/test_object_detection_backends.py`: **5 passed**, with Python 3.12 / PyTorch 2.10.0+cpu / torchvision 0.25.0+cpu / Transformers 5.18.0.dev0 / TorchMetrics PR `f181bd2` / ultrafast-pycocotools 0.1.11.
- The tests cover both argument parsers, unsupported-backend errors, nontrivial overall/per-class mAP/mAR parity, and actual Trainer and Accelerate evaluation of a small randomly initialized DETR for two evaluation cycles. The latter uses four local images in two batches; it needs no checkpoint or dataset downloads.
- `make fix-repo`: **17 checks passed**; `make style`: **5 checks passed**. The test fetcher identifies the focused test plus both existing PyTorch example test files.
- I also ran the existing slow Accelerate object-detection test. It fails before metric construction while loading the published CPPE-5 DETR checkpoint: `DetrConfig.__post_init__` calls `.get()` on `backbone_kwargs=None`. The unmodified upstream no-Trainer script reproduces that same failure. I have not counted this as a passing training test or changed unrelated model configuration code.

Framework-level training speed and peak memory have not been benchmarked. The motivation is to make the evaluator selectable for users who want to measure it on their workloads, without changing the default or adding a core Transformers dependency.

This implementation and proposal were prepared with AI assistance at my direction. I found no overlapping open ultrafast integration PR/issue in this repository. No Transformers PR has been opened; I will wait for scope approval and the TorchMetrics prerequisite.

**Comment by vinitsonawane45** [comment]
Hi @developer0hye, I’d like to work on this integration if the scope is approved.

My understanding is that the change would be limited to:

* `run_object_detection.py`
* `run_object_detection_no_trainer.py`
* optional dependency/documentation updates
* focused backend/argument/evaluation tests

The default `pycocotools` path would remain unchanged, and `ultrafast` would only be available when the required TorchMetrics backend and `ultrafast-pycocotools` versions are supported, with an actionable error otherwise.

Before I start implementation, could you confirm whether this scope is acceptable to the Transformers maintainers and whether you’d be happy for me to work on it?

I’ll follow the existing object-detection example patterns and keep the implementation minimal and isolated.

**Comment by developer0hye** [comment]
Thanks, @vinitsonawane45 — your description matches the proposed scope. I am the proposer, not a Transformers maintainer, and there has not yet been maintainer approval, so I cannot confirm acceptance on their behalf.

The implementation and focused tests are already available at [5258ccf](https://github.com/developer0hye/transformers/commit/5258ccfc4d04031b831ecbf9295be0e5cd1b7469), with the validation results and limitations in the issue body. Reviews or additional test results on that branch would be welcome and would help avoid duplicating the implementation.

The prerequisite [TorchMetrics PR #3500](https://github.com/Lightning-AI/torchmetrics/pull/3500) is still an open draft awaiting scope discussion. Its current CI checks have passed, but the backend is not yet merged or released. As documented above, the tested released TorchMetrics 1.9.0 does not support this backend. I am waiting for Transformers scope approval and the TorchMetrics prerequisite before opening the integration PR; the review-only Git dependency will need to become a released minimum version.

**Comment by ArjunPakhan** [comment]
Hi @developer0hye! Thanks for outlining such a clean proposal and providing the draft verification branch. 

I'm keeping an eye on this issue and will be glad to assist with testing/reviews once Lightning-AI/torchmetrics#3500 is merged and Transformers maintainers approve the scope!
